### ISSUE TITLE
Union Overlapping Volumes simply doesn't work

### ISSUE THREAD

**Issue by S3D-r** [Ultimaker/Cura#15191]
### Application Version

5.2.1

### Platform

Win 10

### Printer

Creality CR-10 Max

### Reproduction steps

Put two bodies on platform, make ém overlap. Select _Union Overlapping Volumes_ .  Select _Remove Mesh Intersection_. Slice. Preview.

### Actual results

There will be an outer wall in the edge of the overlap.

### Expected results

The overlapping volumes are supposed be treated as it was a single body. The correct outcome should be infill.

### Checklist of files to include

- [ ] Log file
- [X] Project file

### Additional information & file uploads

[Overlapping bodies.zip](https://github.com/Ultimaker/Cura/files/11185749/Overlapping.bodies.zip)

**Comment by GregValiant** [comment]
Thanks for the report.
I couldn't get it to work as I expected either.
I exported you model from Cura as an STL and then merged the combined model using MS 3D Builder.  As a workaround this looks acceptable.  I don't know how your real models would work out though.
![image](https://user-images.githubusercontent.com/64202104/230807807-739dfabe-3761-4623-9399-57f794bb3a77.png)

**Comment by S3D-r** [comment]
Yeah, your workaround is commonly known, and sometimes it works, but not always: the workaround may raise other issues.

**Comment by nallath** [comment]
This is intended behavior. See the extra info that the setting guide gives:
![image](https://user-images.githubusercontent.com/3922611/231145602-a3f11f98-2cdd-4118-93d2-c0de6242fc3b.png)

**Comment by GregValiant** [comment]
@nallath I get this with Overlapping Volumes turned off.
![image](https://user-images.githubusercontent.com/64202104/231320399-4eeaff8b-54d2-49eb-a3c9-7aa970ccea1c.png)
and this with it turned on.
![image](https://user-images.githubusercontent.com/64202104/231320474-17e85607-7144-4d33-b9f7-36a9b6c35b61.png)
And this one with Remove Mesh Intersections enabled
![image](https://user-images.githubusercontent.com/64202104/231320541-f9ffb2b3-5330-4ca8-a89a-77080e0ebaa5.png)
Am I missing something?  I don't think I've ever gotten models to slice like either one of those in the Setting Guide.

**Comment by S3D-r** [comment]
> This is intended behavior. See the extra info that the setting guide gives: ![image](https://user-images.githubusercontent.com/3922611/231145602-a3f11f98-2cdd-4118-93d2-c0de6242fc3b.png)

OK... So, Cura is just missing the feature where separate intersecting meshes would be printed as one mesh. That would be useful feature, and many other slicers have it.

**Comment by GregValiant** [comment]
This is in 4.13.1.  The models on the left are separate meshes and the model on the right are the same two models but I merged them in MS 3D Builder.  In both of these images "Alternate Mesh Removal" is turned off.
![image](https://github.com/Ultimaker/Cura/assets/64202104/b28ac711-8aa8-4f54-93c8-3e69229c9f80)

This is the same project in 5.4.0.
![image](https://github.com/Ultimaker/Cura/assets/64202104/cf451952-a337-41b1-8514-ae45756f70cf)

This time I turned on "Alternate Mesh Removal".
4.13.1
![image](https://github.com/Ultimaker/Cura/assets/64202104/00b1598f-9e07-4219-82f5-2a3a366ed740)

5.4.0
![image](https://github.com/Ultimaker/Cura/assets/64202104/0048aceb-7827-4a14-a449-f03cf1ce4367)

With this test both versions produced nearly the same result.  It isn't what I expected though.  With AMR turned on there are skins on every layer of the overlapped models.  I thought that was odd.
